corehq/form_processor/parsers/form.py

```diff
@@ -195,6 +195,15 @@ def apply_deprecation(existing_xform, new_xform, interface=None):
 
     interface = interface or FormProcessorInterface(existing_xform.domain)
 
+    if existing_xform.persistent_blobs:
+        for name, meta in existing_xform.persistent_blobs.items():
+            with existing_xform.fetch_attachment(name, stream=True) as content:
+                new_xform.deferred_put_attachment(
+                    content,
+                    name=name,
+                    content_type=meta.content_type,
+                    content_length=meta.content_length,
+                )
```

Comment:
@snopoke: I'm assuming that attachments do not need to be copied when deprecating SQL forms. Does that sound correct to you? Context: https://github.com/dimagi/commcare-hq/pull/12750/commits/d95b941ce7682d86d70c1841e15a759c37348842